sdc/datatypes/hpat_pandas_series_functions.py

```diff
@@ -105,77 +244,212 @@ def hpat_pandas_series_getitem(self, idx):
     _func_name = 'Operator getitem().'
 
     if not isinstance(self, SeriesType):
-        raise TypingError('{} The object must be a pandas.series. Given: {}'.format(_func_name, self))
+        return None
 
-    if isinstance(idx, types.Integer):
-        def hpat_pandas_series_getitem_idx_integer_impl(self, idx):
-            """
-            **Test**: python -m sdc.runtests sdc.tests.test_series.TestSeries.test_series_iloc1
-            """
+    # Note: Getitem return Series
+    index_is_none = self.index is None or isinstance(self.index, numba.types.misc.NoneType)
+    index_is_number = index_is_none or (self.index and isinstance(self.index.dtype, types.Number))
```

Comment:
Hmm, is this valid? There should be check that self.index is of types.Array type somewhere.
Also it's very confusing to call index_is_number when it can also be None. These are different cases, which may require different handling later. In any case if you want to check for "None or numeric index" just call it that way: index_is_none_or_numeric